tmk_core/protocol/ps2_interrupt.c

```diff
@@ -63,14 +71,63 @@ static inline void pbuf_enqueue(uint8_t data);
 static inline bool pbuf_has_data(void);
 static inline void pbuf_clear(void);
 
+#if defined(PROTOCOL_CHIBIOS)
+//------------------------------------------------
+// LEGACY EXT driver, for external interrupts
+// (with newer chibios we could use palLineEnableEvent and so forth...)
+
+void ps2_interrupt_service_routine(void);
+void extcb(EXTDriver *extp, expchannel_t channel) {
+    ps2_interrupt_service_routine();
+}
+
+/*
+  16 slots, one for each pin...
+  currently configured to listen on external changes on pad A8
+ */
+static EXTConfig extcfg = {
+    {
+        {EXT_CH_MODE_DISABLED, NULL}, //0
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL}, //5
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL}, //10
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL},
+        {EXT_CH_MODE_DISABLED, NULL}
+    }
+};
+static EXTChannelConfig ext_clock_channel_config = {EXT_CH_MODE_FALLING_EDGE | EXT_CH_MODE_AUTOSTART | EXT_MODE_GPIOA, extcb}; // NOTE: hardcoded/limited to Port-A
```

Comment:
Is there a reason this is limited to port A? Should it not be a config option, or work it out from `PS2_LINE_DATA`?

Reply:
thaaats complicated... since for ARM/chibios (in the old version QMK drags around) the software interrupt has to be initialized with a per-port datastructure - this could be coded in a way that is user configurable - but the overhead is IMO not worth it (code-complexity and maintainability wise)

Reply:
Except consumers would have no way to override the behaviour and have to patch core code. Given we have that complexity elsewhere, I cant see why we would accept it here. Even if it was an `PS2_GPIO_PORT` option for now, that would be a start.

Reply:
ok, the deine is a solution (albeit not the nicest :P) - done with 057fb7d20